pipelines/common/vars.yaml

```diff
@@ -1,6 +1,32 @@
+# `Version.BuildNumber` variable is set to a value dependent on the source branch and build reason:
+# - For builds on `main` that are triggered by CI or manually: `1.0.123456`
+# - For all other build configurations:                        `1.0-pre`
+#
+# The resultant value should be set on the pipeline `name` with the `$(Rev:r)` suffix to ensure that
+# build numbers are always unique and not erroneously incremented, even if this value is subsequently
+# overwritten by a `##vso[build.updatebuildnumber]` command elsewhere in the pipeline. e.g.:
+# - `name: $(Version)$(Version.Revision.Separator)$(Version.Revision).$(Rev:r)`
+#
+# **NOTE:** `name: $(Version.BuildNumber).$(Rev:r)` does not resolve `Version.BuildNumber` correctly.
+#
+# Additional commentary may be found at:
+# - https://learn.microsoft.com/en-us/azure/devops/pipelines/process/run-number
 variables:
-  Version: 1.0
-  Version.Revision: $[counter(variables['Version'], 0)]
-  Version.BuildNumber: $(Version).$(Version.Revision)
-  Version.BuildVersion: $(Version.BuildNumber)
-  ShouldDeploy: $[and(eq(variables['Build.SourceBranch'], 'refs/heads/main'), ne(variables['Build.Reason'], 'PullRequest'))]
\ No newline at end of file
+  - name: ShouldDeploy
+    value: $[and(eq(variables['Build.SourceBranch'], 'refs/heads/main'), ne(variables['Build.Reason'], 'PullRequest'))]
+  - name: Version
+    value: 1.0
+  - name: Version.Revision
+    ${{ if eq(variables['ShouldDeploy'], 'true') }}:
```

Comment:
This will always return false. The `if` is done at compile time whereas the variable value is set at runtime.

Reply:
Good spot - I thought I had validated this in a test pipeline but seemingly not until just now!